### DISCUSSION TITLE
Q2 2022 blog post

### DISCUSSION THREAD

**Discussion by MabezDev** [MabezDev/blog#13] [General]
Title

**Comment by jessebraham** [comment]
## espflash

- Versions `1.4.0`, `1.4.1`, `1.5.0`, and `1.5.1` released
    - Version `1.6.0` should be out next week, which will include everything listed below
- Lots of bugfixes
- New `partition-table` subcommand added for converting between CSV/binary, displaying
- Allow more options to be passed through to `cargo` when using `cargo-espflash`
- Added support for flashing from WSL2
- Bootloader and partition table detection when using `esp-idf-sys`-based projects
- New `write-bin-to-flash` subcommand added (works similar to `esptool.py`)
- Support added for the ESP32-C2
- New `serial-monitor` subcommand added (for monitoring without flashing)

## esp-alloc

- Is now a thing
    - https://github.com/esp-rs/esp-alloc

## esp-hal

- New peripheral drivers for `USB_SERIAL_JTAG`, `RMT`, `SYSTIMER`, `EFUSE`, `DAC`, `ADC`
- GPIO errata implemented for ESP32
- Now uses time types from `fugit` where applicable
- Basic clock configuration implemented
- Implemented most `embedded-hal` alpha traits
- Initial support for muli-core applications

**Comment by bjoernQ** [comment]
## esp-wifi

- now an `esp-rs` repository
- support for ESP32
- support for BTLE on ESP32C3 and ESP32

## esp-backtrace
- backtrace on bare-metal

**Comment by SergioGasquez** [comment]
## Devcontainers
- Both `cargo-generate` templates now include a question to add support for devcontainers:
  - [esp-idf-template](https://github.com/esp-rs/esp-idf-template)
  - [esp-template](https://github.com/esp-rs/esp-template)
- Flashing from the container is now supported using [web-flash](https://github.com/bjoernQ/esp-web-flash-server) and Wokwi simulation (including GDB Debugging) with [wokwi-server](https://github.com/MabezDev/wokwi-server) 

## Rust Installer
- Add support for new arguments:
  - `-n|--nightly-version`  chooses `nightly` version for RiscV targets
  - `-s|--esp-idf-version`  installs esp-idf. It will install the passed esp-idf version for the targets specified in `--build-target`
     - `--minified-esp-idf` cleans some unnecessary folders of esp-idf
  - Merges esp-idf export file with the one generated for the xtensa rust toolchain
- CI workflow to generate several tags for [idf-rust](https://hub.docker.com/r/espressif/idf-rust) image
- Installer cleanup
